libr/core/cmd_cmp.c

```diff
@@ -58,103 +58,177 @@ R_API RCoreCmpWatcher *r_core_cmpwatch_get(RCore *core, ut64 addr) {
 	return NULL;
 }
 
-R_API int r_core_cmpwatch_add(RCore *core, ut64 addr, int size, const char *cmd) {
+R_API bool r_core_cmpwatch_add(RCore *core, ut64 addr, int size, const char *cmd) {
 	RCoreCmpWatcher *cmpw;
-	if (size < 1) {
-		return false;
-	}
+	bool found = false;
+	r_return_val_if_fail (core, false);
+	r_return_val_if_fail (cmd, false);
+	r_return_val_if_fail (strlen (cmd) < sizeof (cmpw->cmd), false);
```

Comment:
That cant make release builds safe from segfaults. There shouldnt be a limit imho